core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -95,6 +95,7 @@ private[spark] object CompressionCodec {
   val FALLBACK_COMPRESSION_CODEC = "snappy"
   val DEFAULT_COMPRESSION_CODEC = "lz4"
   val ALL_COMPRESSION_CODECS = shortCompressionCodecNames.values.toSeq
+  val ALL_COMPRESSION_CODECS_SHORT: Set[String] = shortCompressionCodecNames.keySet
```

Comment:
Instead of exposing this and supporting only short codec names for checkpoint, the pattern should be same as in rest of spark code when dealing with codec's.
```
sparkConf.getOption("spark.checkpoint.compress.codec").map(c => 
  logInfo(s"Compressing checkpoint using $c.")
  CompressionCodec.createCodec(conf, c)
).getOrElse(fileStream)

```
This will ensure that support for checkpoint compression is in line with rest of spark (short and long classes, no need to introduce 'none')

Note: you will need to change fileStream to a `lazy val` - so that if codec creation throws exception, we dont leave dangling streams around (with limited block visibility scope to fileStream)